src/credentials/SsiCredentialIssuer.Expiry.App/ExpiryCheckService.cs

```diff
@@ -109,21 +111,30 @@ private static async Task ProcessCredentials(
         IProcessStepRepository<ProcessTypeId, ProcessStepTypeId> processStepRepository,
         CancellationToken cancellationToken)
     {
-        if (data.ScheduleData.IsVcToDelete)
+        try
         {
-            companySsiDetailsRepository.RemoveSsiDetail(data.Id, data.Bpnl, data.RequesterId);
-        }
-        else if (data.ScheduleData.IsVcToDecline)
-        {
-            HandleDecline(data.Id, companySsiDetailsRepository, processStepRepository);
+            if (data.ScheduleData.IsVcToDelete)
+            {
+                companySsiDetailsRepository.RemoveSsiDetail(data.Id, data.Bpnl, data.RequesterId);
+            }
+            else if (data.ScheduleData.IsVcToDecline)
+            {
+                HandleDecline(data.Id, companySsiDetailsRepository, processStepRepository);
+            }
+            else
+            {
+                await HandleNotification(data, companySsiDetailsRepository, portalService, cancellationToken).ConfigureAwait(false);
+            }
+
+            // Saving here to make sure the each credential is handled by there own 
+            await repositories.SaveAsync().ConfigureAwait(ConfigureAwaitOptions.None);
+
         }
-        else
+        catch (Exception ex)
         {
-            await HandleNotification(data, companySsiDetailsRepository, portalService, cancellationToken).ConfigureAwait(false);
+            var message = $"An error occurred while processing credential details '{ex.Message}' for ID '{data.Id}'";
```

Comment:
Can we change log message here to improve exception readability. What do you think?

```suggestion
            var message = $"Failed to process credential with ID '{data.Id}': {ex.Message}";
```